Update Window.postMessage() syntax

achamas-playco opened the pull request:
"targetWindow" variable was missing from primary example. This may cause confusion.

<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
<!-- ✍️ In a sentence or two, describe your changes -->
Include `targetWindow` variable in primary example. This is part of the syntax and is mentioned in the proceeding syntax explanation, though the fact that the variable is missing may cause confusion to unfamiliar developers.

#### Motivation
<!-- ❓ Why are you making this change? Help us understand how your changes help readers. -->
To help improve the clarity and effectiveness of the documentation.

#### Supporting details
<!-- 🔗 Link to supporting information, such as bug trackers, source control, release notes, and vendor docs. -->
Further examples show "targetWindow" - https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#example

#### Related issues
<!-- 🔨 If applicable, use "Fixes #XYZ" -->

#### Metadata
<!-- ✅ Check a box if applicable, like this: [x]

This PR…
-->
- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [✅  ] Fixes a typo, bug, or other error

<!-- 👷‍♀️ After submitting, review the results of the "Checks" tab! -->

achamas-playco committed:
Update index.md

"targetWindow" variable was missing from primary example. This may cause confusion.

teoli2003 commented:
Hi!

I agree that this causes confusion.

The solution is a bit more complex. Our syntax structure is described [here](https://developer.mozilla.org/en-US/docs/MDN/Structures/Page_types/API_method_subpage_template).

What needs to be done is to make it matches the template:
1. Remove the brackets in the syntax (they denote optionality but this is explained by having two separate lines like now). They are useless and confusing.
2. Create the _Parameters_ subsection with the explanation of the parameters (like now)
3. Create a _Return value_ subsection with the content that is now under _`targetWindow`_. The `targetWindow` becomes useless and naturally disappear.

Do you want to give it a try?

achamas-playco commented:
Hi Jean,

I see, please leave this with me and I will take another look at it.

Cheers,
Ali

On Mon, Apr 4, 2022 at 5:17 PM Jean-Yves Perrier ***@***.***>
wrote:

> Hi!
>
> I agree that this causes confusion.
>
> The solution is a bit more complex. Our syntax structure is described here
> <https://developer.mozilla.org/en-US/docs/MDN/Structures/Page_types/API_method_subpage_template>
> .
>
> What needs to be done is to make it matches the template:
>
>    1. Remove the brackets in the syntax (they denote optionality but this
>    is explained by having two separate lines like now). They are useless and
>    confusing.
>    2. Create the *Parameters* subsection with the explanation of the
>    parameters (like now)
>    3. Create a *Return value* subsection with the content that is now
>    under *targetWindow*. The targetWindow becomes useless and naturally
>    disappear.
>
> Do you want to give it a try?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mdn/content/pull/14642#issuecomment-1087201481>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AVXCDSV763QVCVV2AH5JSCTVDKJRVANCNFSM5SONXAQQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

achamas-playco committed:
Update index.md

As per @teoli2003 advice:

1. Remove the brackets in the syntax (they denote optionality but this is explained by having two separate lines like now). They are useless and confusing.
2. Create the Parameters subsection with the explanation of the parameters (like now)
3. Create a Return value subsection with the content that is now under targetWindow. The targetWindow becomes useless and naturally disappear.

Though I must say I'm not 100% sure on the point 3. `postMessage` function does not return a value, did I miss something in the original feedback? Why would we use `targetWindow` for return value here?

achamas-playco commented:
Hi @teoli2003 I made the changes, though are you sure about point 3? `postMessage` does not have a return type (at least according to TypeScript) so why would `targetWindow` be used for that info?

teoli2003 commented:
Of course! I wasn't awake I think. You are right. The part there should be deleted (no reason to explain how to get a Window…) and just add a text "None." under _Return value_.

Sorry about this.

teoli2003 commented on files/en-us/web/api/window/postmessage/index.md:
```diff
@@ -69,6 +60,19 @@ postMessage(message, targetOrigin, [transfer])
 
+### Return value
+
+- `targetWindow`
```
Set it to _None._

Sorry for the misleading information.

achamas-playco replied:
Hi @teoli2003 no problem, I noticed the [Syntax guideline](https://developer.mozilla.org/en-US/docs/MDN/Structures/Syntax_sections) recommends `undefined` so I made that change if that's ok.

achamas-playco committed:
Update index.md

Fix return section, set to `undefined` as recommended in [Syntax guideline](https://developer.mozilla.org/en-US/docs/MDN/Structures/Syntax_sections)

teoli2003 reviewed (approved):
Thanks!

teoli2003 commented:
Thanks a lot!

Looks good to me.

Congratulations upon your first merged commit here! Welcome aboard! :tada:

achamas-playco commented:
Hey great, thanks for your help @teoli2003!
